backend/src/services/api-service/optimal/optimal-service.ts

```diff
@@ -28,6 +31,34 @@ export function findOptimalSetsForMeal(mealName: string, input: SetCoverProducti
   return customOptimalSet(mealName, input, TEAMFINDER_SET_COVER_TIMEOUT, monteCarloIterations);
 }
 
+/**
+ * Finds the optimal Pokemon for a specific ingredient
+ *
+ * API: /api/optimal/ingredient
+ */
+export function findOptimalMonsForIngredient(
+  ingredientName: string,
+  input: SetCoverProductionStats,
+  monteCarloIterations: number
+) {
+  const ingredient: Ingredient = getIngredientForName(ingredientName);
```

Comment:
minor: but no need to explicitly state return type of Ingredient. The compiler will understand it's Ingredient. The discussion regarding using explicit types vs not is a whole topic on its own, but I prioritize matching overall code structure right now

Reply:
Removed the extra annotations. I'm reminded of the "technical details" for [DreamBerd's type annotations](https://github.com/TodePond/DreamBerd?tab=readme-ov-file#types).